Tasks/AzureFunctionDeploymentV1/taskparameters.ts

```diff
@@ -0,0 +1,109 @@
+            }
+            tl.debug(`Resource Group: ${taskParameters.ResourceGroupName}`);
+        }
+        else if(!taskParameters.WebAppKind){
+            var appService = new AzureAppService(taskParameters.azureEndpoint, taskParameters.ResourceGroupName, taskParameters.WebAppName);
+            var configSettings = await appService.get(true);
+            taskParameters.WebAppKind = webAppKindMap.get(configSettings.kind) ? webAppKindMap.get(configSettings.kind) : configSettings.kind;
+        }
+
+        taskParameters.isLinuxApp = taskParameters.WebAppKind && taskParameters.WebAppKind.indexOf("Linux") !=-1;
+
+        var endpointTelemetry = '{"endpointId":"' + taskParameters.connectedServiceName + '"}';
+        console.log("##vso[telemetry.publish area=TaskEndpointId;feature=AzureRmWebAppDeployment]" + endpointTelemetry);
+       
+        taskParameters.Package = new Package(tl.getPathInput('package', true));
+        tl.debug("intially web config parameters :" + taskParameters.WebConfigParameters);
+        if(taskParameters.Package.getPackageType() === PackageType.jar && (!taskParameters.isLinuxApp)) {
+            if(!taskParameters.WebConfigParameters) {
+                taskParameters.WebConfigParameters = "-appType java_springboot";
+            }
+            if(taskParameters.WebConfigParameters.indexOf("-appType java_springboot") < 0) {
+                taskParameters.WebConfigParameters += " -appType java_springboot";
+            }
+            if(taskParameters.WebConfigParameters.indexOf("-JAR_PATH D:\\home\\site\\wwwroot\\*.jar") >= 0) {
+                var jarPath = webCommonUtility.getFileNameFromPath(taskParameters.Package.getPath());
+                taskParameters.WebConfigParameters = taskParameters.WebConfigParameters.replace("D:\\home\\site\\wwwroot\\*.jar", jarPath);
+            } else if(taskParameters.WebConfigParameters.indexOf("-JAR_PATH ") < 0) {
+                var jarPath = webCommonUtility.getFileNameFromPath(taskParameters.Package.getPath());
+                taskParameters.WebConfigParameters += " -JAR_PATH " + jarPath;
+            }
+            if(taskParameters.WebConfigParameters.indexOf("-Dserver.port=%HTTP_PLATFORM_PORT%") > 0) {
+                taskParameters.WebConfigParameters = taskParameters.WebConfigParameters.replace("-Dserver.port=%HTTP_PLATFORM_PORT%", "");  
+            }
+            tl.debug("web config parameters :" + taskParameters.WebConfigParameters);
+        }
+
+        if(taskParameters.isLinuxApp) {
+            taskParameters.RuntimeStack = tl.getInput('runtimeStack', false);
+        }
+
+        taskParameters.DeploymentType = DeploymentType[(tl.getInput('deploymentMethod', false))];
+
+        return taskParameters;
+    }
+}
+
+export enum DeploymentType {
+    auto,
+    zipDeploy,
+    runFromPackage,
+    warDeploy
+}
+
+export interface TaskParameters {
+    connectedServiceName: string;
+    WebAppName?: string;
```

Comment:
Is webAppName can be null ? tl.getInput('appName', true); here it is mandatory field.